flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/planner/delegation/hive/HiveParser.java

```diff
@@ -190,6 +209,12 @@ public List<Operation> parse(String statement) {
             LOG.warn("Current catalog is not HiveCatalog. Falling back to Flink's planner.");
             return super.parse(statement);
         }
+        // check whether it's set command
+        Matcher setCmdMatcher = SET_CMD_PATTERN1.matcher(statement);
+        if (setCmdMatcher.find()) {
+            return Collections.singletonList(
+                    processSetCmd(((HiveCatalog) currentCatalog).getHiveConf(), setCmdMatcher));
+        }
```

Comment:
It seems Hive also supports other kinds of commands, e.g. RESET/COMPILE. How about we reuse the HiveCommand to notify users of unsupported commands.

```
       HiveCommand hiveCommand = HiveCommand.find(statement.split("\\s+"), false);
        if (hiveCommand != null) {
            return Collections.singletonList(
                    processCmd(hiveCommand, statement, (HiveCatalog) currentCatalog));
        }

```

```
    private Operation processCmd(HiveCommand command, String statement, HiveCatalog catalog) {
        if (command != HiveCommand.SET) {
            throw new UnsupportedOperationException();
        }
        return processSetCmd(catalog.getHiveConf(), statement);
    }
```